examples/google-force-safe-search/google-safe-search.rb

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env ruby
+require 'rubydns'
+
+INTERFACES = [
+	[:udp, "0.0.0.0", 5300],
+	[:tcp, "0.0.0.0", 5300],
+]
+
+# Use upstream DNS for name resolution.
+UPSTREAM = RubyDNS::Resolver.new([
+	[:udp, "8.8.8.8", 53],
+	[:tcp, "8.8.8.8", 53]
+])
+
+$R = Resolv::DNS.new
+Name = Resolv::DNS::Name
+IN = Resolv::DNS::Resource::IN
+
+# Start the RubyDNS server
+RubyDNS::run_server(INTERFACES) do
+	@logger.debug!
+  
+  	#match google.com
+  	match(/^google.com$/, IN::A) do |transaction|
```

Comment:
In order to preserve sub-domains such as admin.google.com, docs.google.com etc. 

It may be necessary to do an exact match. Maybe Google handles this? Probably better to only override the specific top level domains for the force safe search?

Reply:
I do like the idea you suggested if that can work. Even if it is just matching the exact domains within an array.